keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+included in the application load balancer's backend pool, should be: the health
+of service proxy.
+
+### Extend `healthCheckNodePort` to class A services
+
+The following addresses [aligning health checks between services](#align-health-checks-between-services)
+
+Given that Kubernetes already defines a port for the load balancer HC to target
+for class B type services, and the fact that we don't want to re-configure the
+HC when the service changes to class A: it would make sense to extend the
+`healthCheckNodePort` field to class A services. Once the service object is
+updated, the existing value would continue to be used.
+
+The name of this field is also rather representative of what it does, which
+would help user comprehension for what is to be expected of this field. With the
+current HC probes defined individually by each cloud provider it takes the user
+some serious spelunking to figure out exactly why a node has been added/removed
+from the LB set. `healthCheckNodePort` has a very narrow scope (compared to
+other HC endpoints in use today, such as: `NodePort` or the Kubelet's healthz
+port). Besides allowing us to document it better and provide the user with a
+better user experience in general, we would also avoiding mixing different
+features depending on specific fields (ex: `NodePort` - which serves as a
+back-bone to a lot of features) and use it as a "dedicated HC intended
+attribute". Given that we would extend it, we also would not need to worry about
+backport/version skew compatibility - which would have be the case should we opt
+for another mechanism which might impact existing class B functionality
+dependent upon `healthCheckNodePort`.
+
+Clearly a big difficulty with doing this will be to make sure that whatever is
+configured for class B services, remains to only be configured for class B
+services - even though both classes of services will have this field.
+
+Doing this would require: changing the API for class A services, change the HCs
+configured by cloud providers to point to the `healthCheckNodePort` for all
+services, and change the service proxies to open ports for all services which
+have this field, but respond differently depending on if the service is class A
+or B. If A: check healthz, if B: check healthz and if there's a local endpoint.
+
+### User Stories (Optional)
+
+#### "Kube version N after KEP" - Extend `healthCheckNodePort` to ETP=cluster
+
+We need to being by performing the necessary changes to have class A services
+get assigned a `healthCheckNodePort`. This means, besides the implementation,
+that we validate that no component in Kubernetes acts solely on this field to
+perform setup for services which are class B. If that is the case: we need to
+change those components too.
+
+#### "Kube version N after KEP" - Implement service proxy logic
+
+Once the API for class A services has changed to include them getting
+`healthCheckNodePort` assigned, we need to update Kube-proxy with the logic
```

Comment:
there are more services proxies than kube-proxy, that should be other criteria, we need to ensure the ecosystem is consistent on their implementation